### PR TITLE
[MTDSA-9445]Updated RAML for incomeTaxAndNics with new object and data items

### PR DESCRIPTION
**Pull request by vfryer12**

### CODE REVIEW

File: resources/public/api/conf/2.0/schemas/selfAssessment/retrieveTaxCalc/retrieve_incomeTaxAndNics_response.json [hunk 1]
```diff
@@ -1771,6 +1879,316 @@
...
+                },
+                "cgtTaxBand": {
+                  "type": "array",
+                  "items": {
+                    "type": "object",
+                    "description": "Details the relevant tax bands for Residential Property and Carried Interest.",
+                    "properties": {
+                      "name": {
+                        "type": "string",
+                        "description": "The name of the tax band. Limited to the following possible values: ",
```
Comment (Mbrookesy) [inline resources/public/api/conf/2.0/schemas/selfAssessment/retrieveTaxCalc/retrieve_incomeTaxAndNics_response.json:2017]: The description doesn't need the 'limited to the following possible values' as the enum field already generates it so currently it's showing it twice.

File: resources/public/api/conf/2.0/schemas/selfAssessment/retrieveTaxCalc/retrieve_incomeTaxAndNics_response.json [hunk 2]
```diff
@@ -1771,6 +1879,316 @@
...
```
Comment (Mbrookesy) [inline resources/public/api/conf/2.0/schemas/selfAssessment/retrieveTaxCalc/retrieve_incomeTaxAndNics_response.json:2142]: The description doesn't need the 'limited to the following possible values' as the enum field already generates it so currently it's showing it twice.

File: resources/public/api/conf/2.0/schemas/selfAssessment/retrieveTaxCalc/retrieve_incomeTaxAndNics_response.json [hunk 3]
```diff
@@ -1771,6 +1879,316 @@
...
+                        "type": "number",
+                        "description": "The rate associated with the tax band. This value must be between 0 and 99.99 up to two decimal places.",
+                        "multipleOf": 0.01,
+                        "minimum": 0,
+                        "maximum": 99.99,
+                        "example": "26.6"
+                      },
+                      "income": {
+                        "type": "number",
+                        "description": "The amount of annual exemption. This value must be between 0 and 99999999999.99 up to 2 decimal places.",
```
Comment (Mbrookesy) [inline resources/public/api/conf/2.0/schemas/selfAssessment/retrieveTaxCalc/retrieve_incomeTaxAndNics_response.json:2158]: Despite what the Spec says, `The amount of annual exemption` doesn't feel like it matches up with the field, maybe something to ask Casey about.

File: resources/public/api/conf/2.0/schemas/selfAssessment/retrieveTaxCalc/retrieve_incomeTaxAndNics_response.json [hunk 4]
```diff
@@ -1771,6 +1879,316 @@
...
+                        "multipleOf": 0.01,
+                        "minimum": 0,
+                        "maximum": 99999999999.99,
+                        "example": "2027.27"
+                      },
+                      "taxAmount": {
+                        "type": "number",
+                        "description": "The amount of annual exemption. This value must be between 0 and 99999999999.99 up to 2 decimal places.",
```
Comment (Mbrookesy) [inline resources/public/api/conf/2.0/schemas/selfAssessment/retrieveTaxCalc/retrieve_incomeTaxAndNics_response.json:2166]: Despite what the Spec says, `The amount of annual exemption` doesn't feel like it matches up with the field, maybe something to ask Casey about.

File: resources/public/api/conf/2.0/schemas/selfAssessment/retrieveTaxCalc/retrieve_incomeTaxAndNics_response.json [hunk 5]
```diff
@@ -1771,6 +1879,316 @@
...
+                        "type": "number",
+                        "description": "The rate associated with the tax band. This value must be between 0 and 99.99 up to two decimal places.",
+                        "multipleOf": 0.01,
+                        "minimum": 0,
+                        "maximum": 99.99,
+                        "example": "20.14"
+                      },
+                      "income": {
+                        "type": "number",
+                        "description": "The amount of annual exemption. This value must be between 0 and 99999999999.99 up to 2 decimal places.",
```
Comment (Mbrookesy) [inline resources/public/api/conf/2.0/schemas/selfAssessment/retrieveTaxCalc/retrieve_incomeTaxAndNics_response.json:2033]: Despite what the Spec says, `The amount of annual exemption` doesn't feel like it matches up with the field, maybe something to ask Casey about.

File: resources/public/api/conf/2.0/schemas/selfAssessment/retrieveTaxCalc/retrieve_incomeTaxAndNics_response.json [hunk 6]
```diff
@@ -1771,6 +1879,316 @@
...
+                        "multipleOf": 0.01,
+                        "minimum": 0,
+                        "maximum": 99999999999.99,
+                        "example": "215.15"
+                      },
+                      "taxAmount": {
+                        "type": "number",
+                        "description": "The amount of annual exemption. This value must be between 0 and 99999999999.99 up to 2 decimal places.",
```
Comment (Mbrookesy) [inline resources/public/api/conf/2.0/schemas/selfAssessment/retrieveTaxCalc/retrieve_incomeTaxAndNics_response.json:2041]: Despite what the Spec says, `The amount of annual exemption` doesn't feel like it matches up with the field, maybe something to ask Casey about.

File: resources/public/api/conf/2.0/schemas/selfAssessment/retrieveTaxCalc/retrieve_incomeTaxAndNics_response.json [hunk 7]
```diff
@@ -1771,6 +1879,316 @@
...
+                }
+              }
+            },
+            "otherGains": {
+              "type": "object",
+              "description": "Details about tax due on Other Gains",
+              "properties": {
+                "gainsIncome": {
+                  "type": "number",
+                  "description": "The amount of gains income from Residential Property and Carried Interest. This value must be between 0 and 99999999999.99 up to 2 decimal places.",
```
Comment (Mbrookesy) [inline resources/public/api/conf/2.0/schemas/selfAssessment/retrieveTaxCalc/retrieve_incomeTaxAndNics_response.json:2072]: The spec and description mention `Residential Property and Carried Interest` despite being in the otherGains object. I feel like it should be `Other Gains` something to mention to Casey

File: resources/public/api/conf/2.0/schemas/selfAssessment/retrieveTaxCalc/retrieve_incomeTaxAndNics_response.json [hunk 8]
```diff
@@ -1771,6 +1879,316 @@
...
+                  "type": "number",
+                  "description": "The amount of gains after losses. This value must be between 0 and 99999999999.99 up to 2 decimal places.",
+                  "multipleOf": 0.01,
+                  "minimum": 0,
+                  "maximum": 99999999999.99,
+                  "example": "221.21"
+                },
+                "attributedGains": {
+                  "type": "number",
+                  "description": "This value must be between 0 and 99999999999.99 up to 2 decimal places.",
```
Comment (Mbrookesy) [inline resources/public/api/conf/2.0/schemas/selfAssessment/retrieveTaxCalc/retrieve_incomeTaxAndNics_response.json:2104]: Currently, the spec has this as TBC, any idea if a description will be made for it?

File: resources/public/api/conf/2.0/schemas/selfAssessment/retrieveTaxCalc/retrieve_incomeTaxAndNics_response.json [hunk 9]
```diff
@@ -1771,6 +1879,316 @@
...
+                  "multipleOf": 0.01,
+                  "minimum": 0,
+                  "maximum": 99999999999.99,
+                  "example": "2022.22"
+                },
+                "netGains": {
+                  "type": "number",
+                  "description": "This value must be between 0 and 99999999999.99 up to 2 decimal places.",
```
Comment (Mbrookesy) [inline resources/public/api/conf/2.0/schemas/selfAssessment/retrieveTaxCalc/retrieve_incomeTaxAndNics_response.json:2112]: Currently, the spec has this as TBC, any idea if a description will be made for it?

File: resources/public/api/conf/2.0/schemas/selfAssessment/retrieveTaxCalc/retrieve_incomeTaxAndNics_response.json [hunk 10]
```diff
@@ -1771,6 +1879,316 @@
...
+                  "multipleOf": 0.01,
+                  "minimum": 0,
+                  "maximum": 99999999999.99,
+                  "example": "2025.25"
+                },
+                "cgtTaxBand": {
+                  "type": "array",
+                  "items": {
+                    "type": "object",
+                    "description": "Details the relevant tax bands for Residential Property and Carried Interest.",
```
Comment (Mbrookesy) [inline resources/public/api/conf/2.0/schemas/selfAssessment/retrieveTaxCalc/retrieve_incomeTaxAndNics_response.json:2138]: The spec and description mention `Residential Property and Carried Interest` despite being in the otherGains object. I feel like it should be `Other Gains` something to mention to Casey

**Review by Mbrookesy — CHANGES_REQUESTED** [review]

File: resources/public/api/conf/2.0/schemas/selfAssessment/retrieveTaxCalc/retrieve_incomeTaxAndNics_response.json (hunk 8 above)
Comment (vfryer12) [reply resources/public/api/conf/2.0/schemas/selfAssessment/retrieveTaxCalc/retrieve_incomeTaxAndNics_response.json:2104]: I ran the TBC's descriptions by Richard and he said they were fine?

File: resources/public/api/conf/2.0/schemas/selfAssessment/retrieveTaxCalc/retrieve_incomeTaxAndNics_response.json (hunk 9 above)
Comment (vfryer12) [reply resources/public/api/conf/2.0/schemas/selfAssessment/retrieveTaxCalc/retrieve_incomeTaxAndNics_response.json:2112]: I ran the TBC's descriptions by Richard and he said they were fine?

File: resources/public/api/conf/2.0/examples/selfAssessment/retrieveIncomeTaxAndNics/retrieve_response.json [hunk 11]
```diff
@@ -16,6 +16,20 @@
       "class4NicsAmount": 5000,
       "totalNic": 5000
     },
+    "capitalGainsTax": {
+      "capitalGainsTax": 6000.99,
```
[the remark this answers is not in the document]
Comment (vfryer12) [reply resources/public/api/conf/2.0/examples/selfAssessment/retrieveIncomeTaxAndNics/retrieve_response.json]: Removed and pushed through

File: resources/public/api/conf/2.0/schemas/selfAssessment/retrieveTaxCalc/retrieve_incomeTaxAndNics_response.json (hunk 1 above)
Comment (vfryer12) [reply resources/public/api/conf/2.0/schemas/selfAssessment/retrieveTaxCalc/retrieve_incomeTaxAndNics_response.json]: Removed and pushed through

File: resources/public/api/conf/2.0/schemas/selfAssessment/retrieveTaxCalc/retrieve_incomeTaxAndNics_response.json (hunk 2 above)
Comment (vfryer12) [reply resources/public/api/conf/2.0/schemas/selfAssessment/retrieveTaxCalc/retrieve_incomeTaxAndNics_response.json]: Removed and pushed through

**Pull request merged**